packages/ui/VirtualList/VirtualListBase.js

```diff
@@ -48,446 +47,322 @@ const itemSizesShape = PropTypes.shape({
-			 * The layout direction of the list.
-			 *
-			 * Valid values are:
-			 * * `'horizontal'`, and
-			 * * `'vertical'`.
-			 *
-			 * @type {String}
-			 * @default 'vertical'
-			 * @public
-			 */
-			direction: PropTypes.oneOf(['horizontal', 'vertical']),
-
-			/**
-			 * Called to get the props for list items.
-			 *
-			 * @type {Function}
-			 * @private
-			 */
-			getComponentProps: PropTypes.func,
-
-			/**
-			 * The array for individually sized items.
-			 *
-			 * @type {Number[]}
-			 * @private
-			 */
-			itemSizes: PropTypes.arrayOf(PropTypes.number),
-
-			/**
-			 * Called when the range of items has updated.
-			 *
-			 * Event payload includes the `firstIndex` and `lastIndex` of the list.
-			 *
-			 * @type {Function}
-			 * @private
-			 */
-			onUpdateItems: PropTypes.func,
-
-			/**
-			 * Number of spare DOM node.
-			 * `3` is good for the default value experimentally and
-			 * this value is highly recommended not to be changed by developers.
-			 *
-			 * @type {Number}
-			 * @default 3
-			 * @private
-			 */
-			overhang: PropTypes.number,
-
-			/**
-			 * When `true`, the list will scroll by page.  Otherwise the list will scroll by item.
-			 *
-			 * @type {Boolean}
-			 * @default false
-			 * @private
-			 */
-			pageScroll: PropTypes.bool,
-
-			/**
-			 * `true` if RTL, `false` if LTR.
-			 *
-			 * @type {Boolean}
-			 * @private
-			 */
-			rtl: PropTypes.bool,
-
-			/**
-			 * The spacing between items.
-			 *
-			 * @type {Number}
-			 * @default 0
-			 * @public
-			 */
-			spacing: PropTypes.number,
-
-			/**
-			 * Called to execute additional logic in a themed component when updating states and bounds.
-			 *
-			 * @type {Function}
-			 * @private
-			 */
-			updateStatesAndBounds: PropTypes.func
-		}
-
-		static defaultProps = {
-			cbScrollTo: nop,
-			dataSize: 0,
-			direction: 'vertical',
-			overhang: 3,
-			pageScroll: false,
-			spacing: 0
-		}
+		let nextState = null;
 
-		constructor (props) {
-			let nextState = null;
+		const containerRef = useRef();
+		const contentRef = useRef();
+		const itemContainerRef = useRef();
 
-			super(props);
+		const [prevChildProps, setPrevChildProps] = useState(null);
+		const [prevFirstIndex, setPrevFirstIndex] = useState(0);
+		const [updateFrom, setUpdateFrom] = useState(0);
+		const [updateTo, setUpdateTo] = useState(0);
+		const [firstIndex, setFirstIndex] = useState(0);
+		const [numOfItems, setNumOfItems] = useState(0);
 
-			this.containerRef = React.createRef();
-			this.contentRef = React.createRef();
-			this.itemContainerRef = React.createRef();
-
-			if (props.clientSize) {
-				this.calculateMetrics(props);
-				nextState = this.getStatesAndUpdateBounds(props);
-			}
-
-			this.state = {
-				firstIndex: 0,
-				numOfItems: 0,
-				prevChildProps: null,
-				prevFirstIndex: 0,
-				updateFrom: 0,
-				updateTo: 0,
-				...nextState
-			};
+		if (props.clientSize) {
+			calculateMetrics(props);
+			nextState = getStatesAndUpdateBounds();
```

Comment:
I'm just curious why the name of this variable is not `statesAndUpdateBounds` since you named like that in other functions.

Reply:
I pursued short name. you're right. I fixed.